flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/factories/HiveFunctionDefinitionFactory.java

```diff
@@ -67,14 +69,26 @@ public HiveFunctionDefinitionFactory(HiveConf hiveConf) {
 		this.hiveShim = HiveShimLoader.loadHiveShim(hiveVersion);
 	}
 
+	public HiveFunctionDefinitionFactory(String hiveVersion) {
+		checkArgument(!StringUtils.isNullOrWhitespaceOnly(hiveVersion),
+			"Hive version cannot be null or empty string");
+		this.hiveVersion = hiveVersion;
+		this.hiveShim = HiveShimLoader.loadHiveShim(hiveVersion);
+	}
+
 	@Override
 	public FunctionDefinition createFunctionDefinition(String name, CatalogFunction catalogFunction) {
 		if (Boolean.valueOf(catalogFunction.getProperties().get(CatalogConfig.IS_GENERIC))) {
 			FunctionDefinitionUtil.createFunctionDefinition(name, catalogFunction);
 		}
 
-		String functionClassName = catalogFunction.getClassName();
+		return createFunctionDefinitionFromHiveFunction(name, catalogFunction.getClassName());
+	}
 
+	/**
+	 * Create a FunctionDefinition from a Hive function's class name.
+	 */
+	public FunctionDefinition createFunctionDefinitionFromHiveFunction(String name, String functionClassName) {
```

Comment:
`public` -> `private`

Reply:
it's used in HiveModule

Reply:
Ah, I see. Sorry for the mistake.

Could you add a note to the javadoc why its public (that it's used in the HiveModule). This method is very Hive specific and has not much to do with the Factory interface itself. Or how about event extracting it to some HiveFunctionUtils class?